From 83cf4a5ac04f9f17cb42e09c02ad963286934809 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Fri, 2 Feb 2018 16:40:24 +0800
Subject: [PATCH 3406/5242] MLK-17514: drm/imx: dcss: directly bypass dec400d
 when no modifier present

commit  935c2f27fd1a5543aa7148f3340cec9285eaecd4 from
https://source.codeaurora.org/external/imx/linux-imx.git

When no modifier present, the 'fb->modifier[0]' may contain
undefined value. So it cannot be used to decide whether the
DEC400D should be set to bypass or not in this case.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-plane.c |   16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-plane.c b/drivers/gpu/drm/imx/dcss/dcss-plane.c
index 6a986c8..eb6e535 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-plane.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-plane.c
@@ -266,14 +266,18 @@ static void dcss_plane_atomic_set_base(struct dcss_plane *dcss_plane)
 
 	switch (plane->type) {
 	case DRM_PLANE_TYPE_PRIMARY:
-		if (modifiers_present) {
-			for (mod_idx = 0; mod_idx < 4; mod_idx++)
-				dcss_dec400d_set_format_mod(dcss_plane->dcss,
-						pix_format,
-						mod_idx,
-						fb->modifier);
+		if (!modifiers_present) {
+			/* No modifier: bypass dec400d */
+			dcss_dec400d_bypass(dcss_plane->dcss);
+			return;
 		}
 
+		for (mod_idx = 0; mod_idx < 4; mod_idx++)
+			dcss_dec400d_set_format_mod(dcss_plane->dcss,
+					pix_format,
+					mod_idx,
+					fb->modifier);
+
 		switch (fb->modifier) {
 		case DRM_FORMAT_MOD_LINEAR:
 		case DRM_FORMAT_MOD_VIVANTE_TILED:
-- 
1.7.9.5

